azure/bucket.ts

```diff
@@ -13,29 +13,124 @@
+        }, { parent: this, protect: preventDestroy });
+        this.container = container;
+
+        this.get = async (key) => {
+            const azStorage = await import("azure-storage");
+            const streamBuffers = await import("stream-buffers");
+
+            const writableStream = new streamBuffers.WritableStreamBuffer();
+
+            const service = new azStorage.BlobService(storageAccount.name.get());
+            await new Promise((resolve, reject) => {
+                service.getBlobToStream(container.name.get(), key, writableStream, err => {
+                    if (err) {
+                        return reject(err);
+                    }
+
+                    resolve();
+                });
+            });
+
+            return writableStream.getContents();
+        };
 
-        throw new Error("Method not implemented.");
+        this.put = async (key, contents) => {
+            const azStorage = await import("azure-storage");
+
+            const service = new azStorage.BlobService(storageAccount.name.get());
+            const readableStream = new ReadableStream(contents);
+
+            await new Promise((resolve, reject) => {
+                service.createBlockBlobFromStream(container.name.get(), key, readableStream, contents.length, err => {
+                    if (err) {
+                        return reject(err);
+                    }
+
+                    resolve();
+                });
+            });
+        };
+
+        this.delete = async (key) => {
+            const azStorage = await import("azure-storage");
+
+            const service = new azStorage.BlobService(storageAccount.name.get());
+            await new Promise((resolve, reject) => {
+                service.deleteBlob(container.name.get(), key, (err: Error) => {
+                    if (err) {
+                        return reject(err);
+                    }
+
+                    resolve();
+                });
+            });
+        };
+
+        this.onPut = async (putName, handler, filter) => {
+            const resourceGroup = await shared.getGlobalResourceGroup();
+            filter = filter || {};
+            serverless.storage.onBlobEvent(putName, storageAccount, (context, buffer) => {
+                    handler({
+                        key: context.bindingData.blobTrigger,
+                        eventTime: context.bindingData.sys.utcNow,
+                        size: buffer.length,
+                    }).then(() => context.done());
+                },
+                {
+                    containerName: container.name,
+                    resourceGroup: resourceGroup,
+                    filterPrefix: filter.keyPrefix,
+                    filterSuffix: filter.keySuffix,
+                },
+                { parent: this });
+        };
+
+        this.onDelete = async (delName, handler, filter) => {
+            throw new Error("Method not implemented.");
```

Comment:
Note: azure does not notify about blob deletes *unless* you use service-grid.  I will file a bug on us adding support for this later.

Reply:
Should we provide a different error message here in the meantime - that this functionality isn't supported on in `@pulumi/cloud-azure`?

Reply:
Sure.